Tools/CMake/AMReXParallelBackends.cmake

```diff
@@ -116,8 +116,10 @@ if (AMReX_HIP)
    # ARCH flags -- these must be PUBLIC for all downstream targets to use,
    # else there will be a runtime issue (cannot find
    # missing gpu devices)
+   # Equivalently, relocatable-device-code (RDC) flags are needed for `extern`
+   # device variable support (for codes that use global variables on device)
    target_compile_options(amrex
       PUBLIC
-      $<$<COMPILE_LANGUAGE:CXX>:-m64 --amdgpu-target=${AMReX_AMD_ARCH}> )
+      $<$<COMPILE_LANGUAGE:CXX>:-m64 --amdgpu-target=${AMReX_AMD_ARCH} -fgpu-rdc> )
```

Comment:
Error as in https://github.com/ROCm-Developer-Tools/HIP/issues/2154

Maybe needs to be passed to link flags, too?

Reply:
Interesting issue persists in link step although `-fgpu-rdc` is in the line:
```
[ 64%] Linking CXX executable Test_MultiBlock_Advection
...
/usr/bin/hipcc -fno-operator-names -O3 -DNDEBUG -fgpu-rdc -pthread CMakeFiles/Test_MultiBlock_Advection.dir/main.cpp.o -o Test_MultiBlock_Advection   -L/usr/lib/gcc/x86_64-linux-gnu/9  -Wl,-rpath,/usr/lib/x86_64-linux-gnu/openmpi/lib:/opt/rocm/hiprand/lib:/opt/rocm/rocrand/lib ../../../Src/libamrex.a -pthread /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi_cxx.so /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so /opt/rocm/hiprand/lib/libhiprand.so.1.1.40101 /opt/rocm/rocrand/lib/librocrand.so.1.1.40101 -lgfortran -lquadmath

ar: two different operation options specified
Use of uninitialized value $HIPLDARCHFLAGS in concatenation (.) or string at /usr/bin/hipcc line 667.
clang-12: error: no such file or directory: '/tmp/hipcciV3mYXHI/libamrex.akBYw'
```

Reply:
Potential refs:.
- https://github.com/ROCm-Developer-Tools/HIP/issues/2154
- https://github.com/ROCmSoftwarePlatform/rccl/pull/260